gpytorch/kernels/white_noise_kernel.py

```diff
@@ -13,16 +13,24 @@ def __init__(self, variances):
         super(WhiteNoiseKernel, self).__init__()
         self.register_buffer("variances", variances)
 
+        # Enforce that variances are batch_size x data_size x task_size
```

Comment:
@Balandat @gpleiss What do you guys think of this pattern? For any parameter we define internally, we always reshape it to have 3 dimensions that denote `batch_size x data_size x task_size`. If a user initializes with a 3 dimensional tensor, we keep it as is. If they pass in a 2 dimensional tensor, we assume it is `data_size x task_size` and unsqueeze a singleton batch dimension. If they pass in a  1 dimensional tensor, we assume it is `data_size` and unsqueeze it to `1 x data_size x 1`.

Reply:
Yeah I like this pattern, I think I had suggested that in an earlier discussion.

Reply:
Great. In that case, I think these are all ready to go. I'll work on some multitask unit tests next.